shell/common/rasterizer.cc

```diff
@@ -231,6 +231,19 @@ sk_sp<SkImage> Rasterizer::ConvertToRasterImage(sk_sp<SkImage> image) {
                               });
 }
 
+fml::TimePoint Rasterizer::DetermineRasterizationTime(LayerTree& layer_tree) {
```

Comment:
Is this the time to start rasterization, and thus usually the time to finish build?

Reply:
This implements the "only one per vsync interval" part of the solution. It is computing "when do we want to start this rasterization" based on whether we have already presented a LayerTree in this interval or not. If it is in the future, we will want to postpone our current task until this calculated time.